static/app/stores/Commiters/CommittersContext.tsx

```diff
@@ -0,0 +1,274 @@
+}
+
+export function CommittersProvider(props: CommittersContextProviderProps) {
+  const contextValue = useReducer(committersReducer, props.initialState ?? {});
+
+  return (
+    <CommittersContext.Provider value={contextValue}>
+      {props.children}
+    </CommittersContext.Provider>
+  );
+}
+
+export function useCommitters(): [CommittersState, React.Dispatch<CommittersAction>] {
+  const context = useContext(CommittersContext);
+
+  if (!context) {
+    throw new Error('useCommitters called outside of CommittersContext.Provider');
+  }
+
+  return context;
+}
+
+interface RequiredWithCommittersProps {
+  event: Event;
+  organization: Organization;
+  project: Project | AvatarProject;
+  group?: Group;
+}
+
+export interface WithCommittersProps {
+  committers: Committer[] | undefined;
+}
+
+export function withCommitters<P extends WithCommittersProps>(
+  Component: React.ComponentType<P>
+) {
+  const wrappedComponent: React.ComponentType<
+    Omit<P, keyof WithCommittersProps> & RequiredWithCommittersProps
+  > = (props): React.ReactElement => {
+    const api = useApi();
+    const [state, dispatch] = useCommitters();
+
+    const key = useMemo(() => {
+      return makeCommitterStoreKey({
+        organizationSlug: props.organization.slug,
+        projectSlug: props.project.slug,
+        eventId: props.event.id,
+      });
+    }, [props.organization.slug, props.project.slug, props.event.id]);
+
+    useEffect(() => {
+      if (!props.group?.firstRelease) {
+        return undefined;
+      }
+
+      if (state[key]?.committers || state[key]?.committersLoading) {
+        return undefined;
+      }
+
+      dispatch({
+        type: 'start loading',
+        payload: {
+          eventId: props.event.id,
+          organizationSlug: props.organization.slug,
+          projectSlug: props.project.slug,
+        },
+      });
+
+      let unmounted = false;
+
+      fetchCommitters(api, {
+        organizationSlug: props.organization.slug,
+        projectSlug: props.project.slug,
+        eventId: props.event.id,
+      })
+        .then(response => {
+          if (unmounted) {
+            return;
+          }
+
+          dispatch({
+            type: 'add committers',
+            payload: {
+              committers: response.committers,
+              eventId: props.event.id,
+              organizationSlug: props.organization.slug,
+              projectSlug: props.project.slug,
+            },
+          });
+        })
+        .catch(() => {
+          if (unmounted) {
+            return;
+          }
+          dispatch({
+            type: 'set error',
+            payload: {
+              eventId: props.event.id,
+              organizationSlug: props.organization.slug,
+              projectSlug: props.project.slug,
+            },
+          });
+        });
+
+      return () => {
+        unmounted = true;
+      };
+    }, [props.group, props.organization.slug, props.project.slug, props.event.id, key]);
```

Comment:
Doesn't the key encapsulate changes in all of the other properties?

Reply:
Yeah, right now it does. I like to add it anyways just in case this changes in the future and we dont end up with some stale key there. I think I'll just remove the memo and recompute it each time tbh, simpler that way

Reply:
@k-fish I wouldnt do that, mostly because we then still need to watch prop changes to update state, which then updates this (I may have misunderstood your comment :D)

Reply:
Hmm true I forgot it was cyclical, I just saw `state[key]` and was erring towards having deps match contents as a rule of thumb. If this doesn't error for this use case, it's fine for now.

Reply:
Ye, state refers to the store state here. We are passing just a part of it down to the components (probably not a pattern we want to encourage)